Frameworks/Foundation/NSCFData.mm

```diff
@@ -46,13 +46,20 @@ - (_Nullable instancetype)initWithBytesNoCopy:(void*)bytes length:(NSUInteger)le
 - (_Nullable instancetype)initWithContentsOfFile:(NSString*)filename options:(NSDataReadingOptions)options error:(NSError**)error {
     CFDataRef tempData;
     SInt32 cfError{};
-    
-    if (CFURLCreateDataAndPropertiesFromResource(nullptr, static_cast<CFURLRef>([NSURL fileURLWithPath:filename]), &tempData, nullptr, nullptr, &cfError)) {
+
+    if (!filename) {
```

Comment:
I prefer early returns for failure conditions as it gets it out of the way quickly. What you suggest is a pattern that can and does to severely nested code. I agree that this function is kinda cluttered though but meh.

---
In reply to: [85214139](https://github.com/Microsoft/WinObjC/pull/1232#discussion_r85214139) [](ancestors = 85214139)